### DISCUSSION TITLE
Built-in CSS support is being disabled due to custom CSS configuration being detected.

### DISCUSSION THREAD

**Discussion by saresh-krish** [vercel/next.js#12230] [Help]
Hi,
I am using NextJs 9.3.4, when i run the app i am facing problem
**"Warning: Built-in CSS support is being disabled due to custom CSS configuration being detected.
See here for more info: https://err.sh/next.js/built-in-css-disabled"**
so in some time external style is not rendering properly and we give to hard reload then the styles work properly .
How to solve this problem??
I need to use external components style for there particular components.
how to use external style CSS and SaSS in **next.config.js** file.

Thanks
Saresh.B

**Comment by saresh-krish** [comment]
Hello
# Bug report

## Describe the bug

Nextjs CSS & SASS in next.config.js

## To Reproduce

I am working on  NextJs 9.3.4, when i run the app i am facing problem
"Warning: Built-in CSS support is being disabled due to custom CSS configuration being detected.
See here for more info: https://err.sh/next.js/built-in-css-disabled"
so in some time external style is not rendering properly and we give to hard reload then the styles work properly .
How to solve this problem??
I need to use external components style for there particular components.
how to use external style CSS and SaSS in next.config.js file.



## Expected behavior

Styles(CSS & SASS) will be work on individual components


## System information

- OS: Ubuntu 18.04.3 LTS
- Browser (if applies): chrome, Mozilla 
- Version of Next.js: 9.3.4
- Version of Node.js: 10.16.3

## Additional context

Some times styles are not loaded and we need to hot reload it is work fine.
https://github.com/zeit/next.js/issues/12244#issue-607685977

**Reply by timneutkens** [reply]
As asked for "Can you provide all your configuration?" You did not include package.json, next.config.js or any other files from your project.

**Reply by saresh-krish** [reply]
hi Timneutkens
I do not have code in git. i share my package json & next.config.js files
`package.json`
`{
  "name": "saresh",
  "version": "1",
  "description": "Vendor Onboarding",
  "main": "index.js",
  "author": "saresh",
  "license": "MIT",
  "private": true,
  "dependencies": {
    "@ant-design/icons": "latest",
    "@zeit/next-css": "^1.0.1",
    "@zeit/next-sass": "^1.0.1",
    "accesscontrol": "^2.2.1",
    "antd": "^4.1.2",
    "bcryptjs": "^2.4.3",
    "body-parser": "^1.19.0",
    "clipboard": "^2.0.4",
    "connect-pg-simple": "^6.1.0",
    "cookie-parser": "^1.4.5",
    "cors": "^2.8.5",
    "draft-js": "^0.11.5",
    "draftjs-to-html": "^0.9.1",
    "express": "^4.17.1",
    "express-session": "^1.16.2",
    "express-winston": "^3.3.0",
    "formik": "^2.1.4",
    "immutable": "^4.0.0-rc.12",
    "jsonwebtoken": "^8.5.1",
    "mkdirp": "^0.5.1",
    "morgan": "^1.9.1",
    "multer": "^1.4.2",
    "next": "^9.3.4",
    "next-fonts": "^1.0.3",
    "next-seo": "^4.4.0",
    "nextjs-fonts": "^0.22.0",
    "node-sass": "^4.13.1",
    "nodemailer-ntlm-auth": "1.0.1",
    "null-loader": "^3.0.0",
    "passport": "^0.4.0",
    "passport-jwt": "^4.0.0",
    "passport-local": "^1.0.0",
    "pg": "^8.0.0",
    "proxy": "1.0.1",
    "proxy-test-server": "1.0.0",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-draft-wysiwyg": "^1.14.5",
    "react-helmet": "^6.0.0",
    "react-icons": "^3.9.0",
    "react-phone-input-2": "^2.13.3",
    "sinon": "9.0.1",
    "smtp-server": "3.6.0",
    "uuid": "^3.4.0",
    "webpack": "^4.43.0",
    "winston": "^3.2.1"
  },
  "scripts": {
    "dev": "node server.js",
    "build": "next build",
    "start": "NODE_ENV=production node server.js"
  },
  "devDependencies": {
    "babel-plugin-import": "^1.12.1",
    "eslint": "^6.1.0",
    "eslint-plugin-react": "^7.14.3"
  },
  "eslintConfig": {
    "extends": "react-app"
  }
}
`

`next.config.js`

`const withCss = require("@zeit/next-css");
const withSass = require("@zeit/next-sass");
const withFonts = require("next-fonts");

module.exports = withCss(
  withSass(
    withFonts({
      webpack: (config, { isServer }) => {
        if (isServer) {
          const antStyles = /antd\/.*?\/style\/css.*?/;
          const origExternals = [...config.externals];
          config.externals = [
            (context, request, callback) => {
              if (request.match(antStyles)) return callback();
              if (typeof origExternals[0] === "function") {
                origExternals[0](context, request, callback);
              } else {
                callback();
              }
            },
            ...(typeof origExternals[0] === "function" ? [] : origExternals)
          ];
          config.module.rules.push({
            test: /\.svg$/,
            use: ["@svgr/webpack"]
          },
          {
            test: /\.(png|jpg|gif|svg|eot|otf|ttf|woff|woff2)$/,
            use: {
              loader: "url-loader"
            }
          }

          );
          config.module.rules.unshift({
            test: antStyles,
            use: "null-loader"
          });
     
        }
        return config;
      },
      serverRuntimeConfig: { 
        NODE_ENV: process.env.NODE_ENV,
      }, 

      publicRuntimeConfig: {
         NS_API: process.env.NS_API
      }
    })
  )
);`

my main problem is external component based styles is not loading properly 
Ex:
- -Test Folder
--------index.jsx
-------- style.css

I have include the 'stlye.css' into 'index.jsx'. When i run app first time the 'style.css' is not loading. In same as 'Sass' file also.


Thanks
B.Saresh

**Comment by ebs-PriyankaKanwar** [comment]
we have updated next JS, all things are working fine except one the CSS is not loaded in proper order in production mode, however, it's working fine in dev mode.

Here are the versions
Node: v13.7.0
React:^16.12.0
NextJs:9.1.6
OS: Linux - ubuntu

Here is what I got when did build:

├ static/pages/_app.js 16.1 kB
├ chunks/1899c3d08f1d4a9ab6e4e22050e66cd1600464e8.b9ea19.js 3.53 kB
├ chunks/48183758.ede116.js 65 B
├ chunks/57a5805a715b88beb8aec3ac43d8f962ef8879cc.53c4e4.js 2.34 kB
├ chunks/9ed67fcaced240be487e7a2880bfff868999eacb.76e83a.js 8.29 kB
├ chunks/ab68c4ca07cacf4e3048ee2d6f69810106b4b489.68a8b3.js 10.4 kB
├ chunks/d32b6132cd0996134c9f87f7274f553f791cd563.18dde5.js 6.39 kB
├ chunks/d6e4d54ae3ad4e326943499c7720eb830c61721b.b7693f.js 4.41 kB
├ chunks/framework.4dd100.js 40.3 kB
├ runtime/main.d3f0e8.js 13.2 kB
├ runtime/webpack.361a6a.js 1.59 kB
├ css/16c4518f007189356065.css 54.9 kB
└ css/f93669f8200e9712e425.css 30.6 kB

Here is my next.config :
const withFonts = require('next-fonts');
const withImages = require('next-images');
// const withCSS = require('@zeit/next-css')

const withPlugins = require('next-compose-plugins');
const nextEnv = require('next-env');
const dotenvLoad = require('dotenv-load');

dotenvLoad();

withFonts({
webpack(config, options) {
// config.plugins.push(new webpack.EnvironmentPlugin(localEnv))
return config;
}
})

module.exports = withPlugins([

nextEnv({
staticPrefix: 'CUSTOM_STATIC_',
publicPrefix: 'CUSTOM_PUBLIC_',
}),withFonts, withImages
]);

I have two things here
1 - why we are having two files of CSS when created build? How can we merge into one CSS file?
2 - How I can fix it so that it will work in the same manner as in dev mode?

Any suggestions will be helpful.

**Comment by saresh-krish** [comment]
**Hi Timneutkens**

Any updated for this problem solution.

**Thanks
B.Saresh**
